### PR TITLE
feat(split-button): surface styled system margin props FE-3559

### PR DESCRIPTION
**Pull request by edleeks87**
### Proposed behaviour

Surfaces `styled-system/margin` props to allow consumers to adjust spacing of `SplitButton`
Surfaces `styled-system/margin` props to allow consumers to adjust spacing of `MultiActionButton`

### Current behaviour

No margin props surfaced on either component

### Checklist

- [x] Commits follow our style guide
- [x] Screenshots are included in the PR if useful
- [x] All themes are supported if required
- [x] Unit tests added or updated if required
- [x] Cypress automation tests added or updated if required
- [x] Storybook added or updated if required
- [x] Typescript `d.ts` file added or updated if required
- [x] Carbon implementation and Design System documentation are congruent

### Additional context

`d.ts` files have been left intentionally as there is an active PR to address all of them in carbon

### Testing instructions

https://codesandbox.io/s/carbon-quickstart-forked-n85nz

### CODE REVIEW

File: src/components/multi-action-button/multi-action-button.component.js [hunk 1]
```diff
@@ -71,6 +71,12 @@ class MultiActionButton extends SplitButton {
 }
 
 MultiActionButton.propTypes = {
+  /** The additional button to display. */
```
Comment (sjohnson-sage) [inline src/components/multi-action-button/multi-action-button.component.js:74]: This should have the margin props in the `propTypes` here, see https://github.com/Sage/carbon/blob/master/src/components/icon-button/icon-button.component.js#L42 for an example

Comment (sjohnson-sage) [reply src/components/multi-action-button/multi-action-button.component.js:74]: and what you've done in split button

Comment (sjohnson-sage) [reply src/components/multi-action-button/multi-action-button.component.js:74]: Also the margin props aren't actually getting passed to `StyledMultiActionButton` anywhere

**Review by sjohnson-sage — COMMENTED** [review]
Needs some changes to `MultiActionButton` mate

File: src/components/multi-action-button/multi-action-button.component.js (hunk 1 above)
Comment (edleeks87) [reply src/components/multi-action-button/multi-action-button.component.js:74]: The prop types come from `SplitButton` which MultiActionButton extends, if they are passed to the `StyledMultiActionButton` then they are set twice. For now I think this is the right approach and when they are refactored to remove the extension we can then spread them on to the root

**Comment by edleeks87** [comment]
Closed as we need to refactor `MultiActionButton` to remove the extension of `SplitButton`

**Pull request closed without merging**